openedx/openedx-platform
# Introduce LMS_INTERNAL_ROOT_URL.

The LMS and edx-enterprise currently communicate via REST APIs, even though they are all running within the same service.

Currently API URLs are constructed using `LMS_ROOT_URL`, but that is problematic on some systems where the external `LMS_ROOT_URL` is not accessible from the private network, or routing internal requests via the external URL is not desirable.

This patch introduces a new setting variable `LMS_INTERNAL_ROOT_URL` which defaults to the value of `LMS_ROOT_URL` for backwards compatibility.

The LMS and edx-enterprise will use `LMS_INTERNAL_ROOT_URL` to construct API URLs instead of `LMS_ROOT_URL`.

**Sandbox URLs**:

- LMS: https://pr16510.sandbox.opencraft.hosting/
- Studio: https://studio-pr16510.sandbox.opencraft.hosting/

**Testing Instructions**:

1. Install this patch.
1. Install https://github.com/edx/edx-enterprise/pull/240 and verify that nothing changes by default - enterprise APIs should work just like they did before.
2. Add an entry to your hosts file that maps the domain used in `LMS_ROOT_URL` (change it to something else if using `localhost`) to a non-reachable IP.
3. Verify that interactions involving enterprise APIs don't work.
4. Set `EDXAPP_LMS_INTERNAL_ROOT_URL` to `http://localhost:8000` in your `lms.env.json` file.
5. Verify that enterprise API interactions work correctly again.

**Reviewers**:

- @UmanShahzad 
- edX (TBD)

**TODO**:

- [x] Switch to new `edx-enterprise` release once https://github.com/edx/edx-enterprise/pull/240 gets merged.
